pkg/querier/querier.go

```diff
@@ -258,49 +258,38 @@ func (q querier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Mat
 
 	userID, err := user.ExtractOrgID(ctx)
 	if err != nil {
-		return nil, nil, promql.ErrStorage{Err: err}
+		return storage.ErrSeriesSet(promql.ErrStorage{Err: err})
 	}
 
 	tombstones, err := q.tombstonesLoader.GetPendingTombstonesForInterval(userID, model.Time(sp.Start), model.Time(sp.End))
 	if err != nil {
-		return nil, nil, promql.ErrStorage{Err: err}
+		return storage.ErrSeriesSet(promql.ErrStorage{Err: err})
 	}
 
 	if len(q.queriers) == 1 {
-		seriesSet, warning, err := q.queriers[0].Select(true, sp, matchers...)
-		if err != nil {
-			return nil, warning, err
-		}
+		seriesSet := q.queriers[0].Select(true, sp, matchers...)
```

Comment:
This changes the logic slightly, but end result looks to be the same.

Reply:
I checked the code and practically there should be no difference. The `seriesSet.Err()` had to be handled even before.